src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -303,6 +303,48 @@ protected HttpRequest() {}
         return new HttpRequestBuilderImpl(uri);
     }
 
+    /**
+     * Creates a {@code Builder} seeded from a {@code HttpRequest}.
+     *
+     * <p> This method can be used to build a new request equivalent to the
+     * given request, but with some parts of its state altered.
+     *
+     * @param request the request
+     * @return a new request builder
+     * @throws IllegalArgumentException if a new builder cannot be seeded from
+     *         the given request (for instance, if the request contains illegal
+     *         parameters)
+     * @since TBD
+     */
+    public static Builder newBuilder(HttpRequest request) {
+        Objects.requireNonNull(request);
+        try {
+            final HttpRequest.Builder builder = HttpRequest.newBuilder();
+            builder.uri(request.uri());
+            builder.expectContinue(request.expectContinue());
+            request.headers().map().forEach((name, values) -> values.forEach(value -> builder.header(name, value)));
+
+            request.version().ifPresent(builder::version);
+            request.timeout().ifPresent(builder::timeout);
+            var method = request.method();
+            request.bodyPublisher().ifPresentOrElse(
+                    bodyPublisher -> builder.method(method, bodyPublisher),
+                    () -> {
+                        switch (method) {
+                            case "GET" -> builder.GET();
+                            case "DELETE" -> builder.DELETE();
+                            default -> builder.method(method, HttpRequest.BodyPublishers.noBody());
+                        }
+                    }
+            );
```

Comment:
I know what this piece of code does, and why it is necessary, but it would be good to add some `//` comments above to explain what is going on for the benefice of future maintainers

Reply:
Comment added to clarify ifPresentOrElese statment. See commit https://github.com/openjdk/jdk/pull/1059/commits/7928083ae954c44b5357064b86af0e1a3f53588b